component/src/main/java/io/siddhi/extension/execution/reorder/KSlackExtension.java

```diff
@@ -106,40 +111,34 @@
     private Scheduler scheduler;
     private long lastScheduledTimestamp = -1;
     private ReentrantLock lock = new ReentrantLock();
-//    private int totalCount;
-//    private int printcount;
+
+    private List<Attribute> attributes = new ArrayList<>();
+    private SiddhiAppContext siddhiAppContext;
 
     @Override
     public void start() {
-        //Do nothing
+        if (lastScheduledTimestamp < 0) {
+            lastScheduledTimestamp = this.siddhiAppContext.getTimestampGenerator().currentTime() + timerDuration;
```

Comment:
lastScheduledTimestamp, eventTreeMap, expiredEventTreeMap and other states that get modified need to be moved into the state. 
That's a bug in the current code

Reply:
Fixed the issue in 2951106.